pdfviewer-library/src/main/java/xyz/guutong/androidpdfviewer/PdfViewActivity.java

```diff
@@ -98,93 +99,35 @@ protected void onCreate(Bundle savedInstanceState) {
-        if (showCloseButton)
-            menu.add(0, 1, MENU_CLOSE, R.string.close)
-                    .setIcon(R.drawable.ic_close)
-                    .setShowAsAction(MenuItem.SHOW_AS_ACTION_IF_ROOM);
-        return super.onPrepareOptionsMenu(menu);
-    }
-
-    @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
-        int i = item.getItemId();
-        if (i == MENU_CLOSE) {
-            finish();
-        } else if (i == MENU_SHARE) {
-
-            if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
-                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, 200);
-            } else {
-                saveFile();
-            }
+    private void checkPermissions() {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, 200);
+        } else {
+            getMyPdf();
         }
-        return true;
     }
 
     @Override
     public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
         switch (requestCode) {
             case 200: {
                 if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                    saveFile();
+                    getMyPdf();
                 }
                 return;
             }
         }
     }
 
-    private void saveFile(){
-        AlertDialog.Builder builder = new AlertDialog.Builder(this);
-
-        try {
-            int bytesum = 0;
-            int byteread = 0;
-            File oldfile = new File(getCacheDir(), FileUtil.extractFileNameFromURL(pdfUrl));
-
-            InputStream inStream = new FileInputStream(oldfile.getAbsoluteFile());
-            FileOutputStream fs = new FileOutputStream(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) + "/" + FileUtil.extractFileNameFromURL(pdfUrl));
-            byte[] buffer = new byte[1444];
-            while ((byteread = inStream.read(buffer)) != -1) {
-                bytesum += byteread;
-                fs.write(buffer, 0, byteread);
-            }
-            inStream.close();
-            fs.close();
-
-            builder.setMessage("File was saved in Downloads");
-
-        } catch (Exception e) {
-            builder.setMessage("Unable to save file.");
-        }
-
-        builder.show();
-    }
-
-    @Override
-    public void onSuccess(String url, String destinationPath) {
-        File pdf = new File(destinationPath);
-
-        pdfView.fromFile(pdf)
+    private void getMyPdf() {
+        // Log.d("pdfUrl", pdfUrl);
+        // loading local pdf file
+        File file = new File(pdfUrl);
+        pdfView.fromFile(file)
```

Comment:
are we sure that file is not null?
do we have a file.close() ?
can we add a try carch finally statement ?